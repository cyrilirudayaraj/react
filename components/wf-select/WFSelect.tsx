import React from 'react';
import { Label, Select, FormError } from '@athena/forge';
import { Field, ErrorMessage } from 'formik';
import { Error } from '../error/Error';

const WFSelect = (props: any): JSX.Element => {
  const fieldProps: any = {
    id: props.name,
    name: props.name,
    options: props.options,
    disabled: props.disabled,
    required: props.required === undefined ? true : props.required,
  };
  if (props.onChange) {
    fieldProps.onChange = props.onChange;
  }
  return (
    <div className="select-wrapper">
      <Label text={props.label} className="fe_u_margin--top-medium" />
      <Field
        as={Select}
        {...fieldProps}
        className="fe_u_margin--top-xsmall fe_u_margin--bottom-medium"
      />
      {props.showError === undefined ? (
        <ErrorMessage name={props.name} component={Error} />
      ) : props.showError ? (
        <div>
          <FormError>{props.errorMessage}</FormError>
        </div>
      ) : null}
    </div>
  );
};

export default WFSelect;
