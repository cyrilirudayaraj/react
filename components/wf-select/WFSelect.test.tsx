import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WFSelect from './WFSelect';

configure({ adapter: new Adapter() });
let wrapper: any;

const options = [
  {
    text: 'Apple',
    value: 'apple',
  },
  {
    text: 'Orange',
    value: 'orange',
  },
  {
    text: 'Banana',
    value: 'banana',
  },
];

describe('snapshot test <WFSelect>', () => {
  it('should render as expected', () => {
    wrapper = shallow(
      <WFSelect label="Fruits" name="fruit" options={options} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render with other', () => {
    wrapper = shallow(
      <WFSelect
        label="Fruits"
        name="fruit"
        options={options}
        required={false}
        showError={true}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render with show as false', () => {
    wrapper = shallow(
      <WFSelect
        label="Fruits"
        name="fruit"
        options={options}
        required={false}
        showError={false}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
