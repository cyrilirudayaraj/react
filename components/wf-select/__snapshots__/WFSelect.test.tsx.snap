// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFSelect> should render as expected 1`] = `
"<div className=\\"select-wrapper\\">
  <Label text=\\"Fruits\\" className=\\"fe_u_margin--top-medium\\" required={false} useLegend={false} />
  <Field as={{...}} id=\\"fruit\\" name=\\"fruit\\" options={{...}} disabled={[undefined]} required={true} className=\\"fe_u_margin--top-xsmall fe_u_margin--bottom-medium\\" />
  <FormikConnect(ErrorMessageImpl) name=\\"fruit\\" component={[Function: Error]} />
</div>"
`;

exports[`snapshot test <WFSelect> should render with other 1`] = `
"<div className=\\"select-wrapper\\">
  <Label text=\\"Fruits\\" className=\\"fe_u_margin--top-medium\\" required={false} useLegend={false} />
  <Field as={{...}} id=\\"fruit\\" name=\\"fruit\\" options={{...}} disabled={[undefined]} required={false} className=\\"fe_u_margin--top-xsmall fe_u_margin--bottom-medium\\" />
  <div>
    <FormError alertType=\\"attention\\" />
  </div>
</div>"
`;

exports[`snapshot test <WFSelect> should render with show as false 1`] = `
"<div className=\\"select-wrapper\\">
  <Label text=\\"Fruits\\" className=\\"fe_u_margin--top-medium\\" required={false} useLegend={false} />
  <Field as={{...}} id=\\"fruit\\" name=\\"fruit\\" options={{...}} disabled={[undefined]} required={false} className=\\"fe_u_margin--top-xsmall fe_u_margin--bottom-medium\\" />
</div>"
`;
